test/_debian_11.Dockerfile

```diff
@@ -0,0 +1,17 @@
+FROM buildpack-deps:bullseye-scm
+
+ENV GITDIR /etc/.pihole
+ENV SCRIPTDIR /opt/pihole
+
+RUN mkdir -p $GITDIR $SCRIPTDIR /etc/pihole
+ADD . $GITDIR
+RUN cp $GITDIR/advanced/Scripts/*.sh $GITDIR/gravity.sh $GITDIR/pihole $GITDIR/automated\ install/*.sh $SCRIPTDIR/
+ENV PATH /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:$SCRIPTDIR
+
+RUN true && \
+    chmod +x $SCRIPTDIR/*
+
+ENV PH_TEST true
+ENV OS_CHECK_DOMAIN_NAME dev-supportedos.pi-hole.net
```

Comment:
Reminder that this will need to be removed/changed before relesae. (I think?)

Reply:
Thinking about this more. It would probably be useful to merge this PR as-is with the override for `dev-supportedos.pi-hole.net`. That way the PRs can start running tests against Bullseye - since we know we need to support it - but we also won't be blocked on 'officially' supporting it yet by updating the DNS entry for `supportedos.pi-hole.net`.

Once its decided to officially support it, then the DNS entry will have to be updated, and someone can submit another PR to disable this dev override. Until then, no harm to keep it set to dev, and we get the added benefit of extra tests.

Reply:
Yeah, actually you are right.

I started thinking about this the other day, and then found a different thing to think about. But as you say, no harm in the tests looking at what's on `dev-supportedos.pi-hole.net`. Actually, might even be preferable for all the tests Docker files to have that override do you think?

The tests are only ever run on branches based on `development`, and allows us to continuously try out new OS's without updating the main DNS record.

Reply:
The only time I can think of it being useful not to test dev is a release merge to master- in which we want to test the prod settings to make sure it's updated with what we are trying to support

Reply:
Though `versions.pi-hole.net` is unlikely to get updated until _after_ the new stuff is released, anyway

Reply:
If that's the case, then the added complexity isn't worth it

Reply:
> Actually, might even be preferable for all the tests Docker files to have that override do you think?

I'm on board with this